homeassistant/components/uptimerobot/config_flow.py

```diff
@@ -17,41 +21,61 @@
 STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required(CONF_API_KEY): str})
 
 
-async def validate_input(hass: HomeAssistant, data: ConfigType) -> UptimeRobotAccount:
-    """Validate the user input allows us to connect."""
-    uptime_robot_api = UptimeRobot(data[CONF_API_KEY], async_get_clientsession(hass))
-
-    try:
-        response = await uptime_robot_api.async_get_account_details()
-    except UptimeRobotException as exception:
-        raise CannotConnect(exception) from exception
-    else:
-        if response.status == API_ATTR_OK:
-            return response.data
-        raise CannotConnect(response.error.message)
-
-
 class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Uptime Robot."""
 
     VERSION = 1
 
+    async def _validate_input(
+        self, data: ConfigType
+    ) -> tuple[dict[str, str], UptimeRobotAccount | None]:
+        """Validate the user input allows us to connect."""
+        errors: dict[str, str] = {}
+        response: UptimeRobotApiResponse | UptimeRobotApiError | None = None
+        uptime_robot_api = UptimeRobot(
+            data[CONF_API_KEY], async_get_clientsession(self.hass)
+        )
+
+        try:
+            response = await uptime_robot_api.async_get_account_details()
+        except UptimeRobotException as exception:
+            LOGGER.error(exception)
+            errors["base"] = "cannot_connect"
+        except Exception as exception:  # pylint: disable=broad-except
+            LOGGER.error(exception)
+            errors["base"] = "unknown"
+        else:
+            if response.status != API_ATTR_OK:
+                if response.error.message in (
+                    "api_key not found.",
+                    "api_key is invalid.",
+                ):
+                    errors["base"] = "invalid_api_key"
+                else:
+                    errors["base"] = "unknown"
+                    LOGGER.error(response.error.message)
+
+        account: UptimeRobotAccount | None = (
+            response.data
+            if response and response.data and response.data.email
+            else None
+        )
+        if account:
+            await self.async_set_unique_id(str(account.user_id))
+            self._abort_if_unique_id_configured()
```

Comment:
We're missing a test for abort if already configured. Add a `MockConfigEntry` to simulate an existing entry.